h/db/__init__.py

```diff
@@ -101,6 +101,25 @@ def close_the_sqlalchemy_session(_request):
     return session
 
 
+def _replica_session(request):  # pragma: no cover
+    engine = create_engine(request.registry.settings["sqlalchemy.replica.url"])
```

Comment:
The `sqlalchemy.replica.url` setting is marked as optional when the application boots. What should happen in code paths that use `request.db_replica` if not set? Should this fall back to using a connection to the same DB as `request.db`? Should `request.db_replica` be `None`? Should a standard exception of some kind be generated?

Reply:
Good question, my thoughts:

- I don't think there's should be fallback, if the code explicitly says "use the replica" it should fail if not available/configured.

- This PR doesn't make it a mandatory setting to avoid having to sync-up config in the EBS environment with the deploy.


My suggestion would be:

- Depoy this PR as it is, set the setting in each environment testing it with the /_status endpoint.

- In a follow up PR start using this in the code base: https://github.com/hypothesis/h/blob/main/h/services/bulk_api/group.py#L46

- Once that's deploy mark this setting as required. Our setting/env vars handling is not perfect but having the server stop starting is probably better than having some random feature stop working when creating new enviorments.

Reply:
> This endpoint it's only used from the LMS admin pages now:

The code you linked doesn't mention `db_replica`. Are you saying that you intend to change that code so that it does use the replica?

Reply:
Yes, in a follow up PR. The link to the endpoint there made that tricky to read.